go/exorelay/exorelay_test.go

```diff
@@ -29,12 +32,18 @@ func FeatureContext(s *godog.Suite) {
 	})
 
 	s.Step(`^ExoRelay connects to Exocom$`, func() error {
-		exoInstance.Connect()
-		return nil
+		return exoInstance.Connect()
 	})
 
 	s.Step(`^it registers by sending the message "([^"]*)" with payload:$`, func(expectedName string) error {
-		messages := exocom.ReceivedMessages
+		var messages []structs.Message
+		for {
+			time.Sleep(time.Duration(10) * time.Millisecond)
+			messages = exocom.ReceivedMessages
+			if len(messages) == 1 {
+				break
+			}
+		}
```

Comment:
Maybe this functionality can go into exocom mock? Maybe a `waitForReceivedMessage` that returns a message, error and times out after a period and returns an error